app/ui-react/packages/auto-form/src/widgets/TextButton.tsx

```diff
@@ -1,8 +1,10 @@
 import * as React from 'react';
 
+import { toValidHtmlId } from "./helpers";
 import './TextButton.css';
 
 export interface ITextButtonProps {
```

Comment:
We can extend the `TextButton` to inherit props that are valid for the underlying component, like this `export interface ITextButtonProps extends React.HTMLAttributes<HTMLElement>`. That should allow all `data` attributes to be passed to the component.